test/newsimplification/Utils.spec.ts

```diff
@@ -0,0 +1,65 @@
+
+import { CommonModel } from '../../src/models/CommonModel';
+import { simplifyName, inferTypeFromValue, isModelObject } from '../../src/newsimplification/Utils';
+
+/**
+ * Some of these test are purely theoretical and have little if any merit 
+ * on a JSON Schema which actually makes sense but are used to test the principles.
+ */
+describe('utils', function () {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  })
+  afterAll(() => {
+    jest.restoreAllMocks();
+  })
+  describe('isModelObject', function () {
+    test('should return true if model type is object', function () {
+      const model = new CommonModel();
+      model.type = 'object';
+      expect(isModelObject(model)).toEqual(true);
+    });
+    test('should return true if model type contains object', function () {
+      const model = new CommonModel();
+      model.type = ['object', 'string'];
+      expect(isModelObject(model)).toEqual(true);
+    });
+    test('should return false if contains all types', function () {
+      const model = new CommonModel();
+      model.type = ['object', 'string', 'number', 'array', 'boolean', 'null', 'integer'];
+      expect(isModelObject(model)).toEqual(false);
+    });
+    test('should return false if type is not defined', function () {
+      const model = new CommonModel();
```

Comment:
I prefer to see AAA pattern(rule?) in tests.

Reply:
Actually, I agree, gonna refactor it.